### PR TITLE
books: architecture skeleton

### PR DESCRIPTION
**Pull request by hinto-janai**
### What
Creates the skeleton for the architecture book, with some sections filled in.

- Upstreamed from: https://github.com/hinto-janai/cuprate-architecture
- Live example: https://hinto-janai.github.io/cuprate-architecture

### CODE REVIEW

File: books/architecture/src/foreword.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Foreword
```
Comment (hinto-janai) [inline books/architecture/src/foreword.md:1]: Mostly inspired from [inspirational foreword in the Rust book](https://doc.rust-lang.org/book/foreword.html).

Too pretentious? Not pretentious enough? Perfect?!?

File: books/architecture/src/SUMMARY.md [hunk 2]
```diff
@@ -1,3 +1,124 @@
 # Summary
```
Comment (hinto-janai) [inline books/architecture/src/SUMMARY.md:1]: The 🟡 sections may be worth giving a small review.

File: books/architecture/src/cuprate-architecture.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+Sections are notated with colors indicating how complete they are:
+
+| Color | Meaning |
+|-------|---------|
+| ⚪️    | Empty
+| 🔴    | Severely lacking information
+| 🟠    | Lacking some information
+| 🟡    | Almost ready
+| 🟢    | OK
```
Comment (hinto-janai) [inline books/architecture/src/cuprate-architecture.md:14]: Btw, no sections in this PR are 🟢 yet.

File: books/architecture/src/SUMMARY.md [hunk 4]
```diff
@@ -1,3 +1,124 @@
...
 
-- [TODO](todo.md)
+[Cuprate Architecture](cuprate-architecture.md)
+[🟡 Foreword](foreword.md)
+
+---
+
+- [🟠 Intro](intro/intro.md)
+	- [🟡 Who this book is for](intro/who-this-book-is-for.md)
```
Comment (Boog900) [inline books/architecture/src/SUMMARY.md:9]: The indenting in this page is non-standard, intentional?

File: books/architecture/src/foreword.md (hunk 1 above)
Comment (Boog900) [reply books/architecture/src/foreword.md:1]: Right on the edge :) 

I am ok with it though

File: books/architecture/src/SUMMARY.md (hunk 4 above)
Comment (hinto-janai) [reply books/architecture/src/SUMMARY.md]: Do you mean the tabs? `rustfmt` doesn't do markdown files I guess. Changed to 4 spaces.

**Review by Boog900 — APPROVED** [review]

**Pull request merged**